core/aws_ddk_core/resources/_dms.py

```diff
@@ -0,0 +1,367 @@
+# Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from typing import Any, Dict, List, Optional, Union
+
+import aws_cdk.aws_dms as dms
+from aws_cdk.aws_iam import PolicyStatement, Role, ServicePrincipal
+from aws_ddk_core.config import Config
+from aws_ddk_core.resources.commons import BaseSchema
+from constructs import Construct
+
+_logger: logging.Logger = logging.getLogger(__name__)
+
+
+class DMSEndpointConfiguration(BaseSchema):
+    """DDK DMS Endpoint Marshmallow schema."""
+
+
+class DMSReplicationTaskConfiguration(BaseSchema):
+    """DDK DMS ReplicationTask Marshmallow schema."""
+
+
+class DMSReplicationInstanceConfiguration(BaseSchema):
+    """DDK DMS ReplicationInstance Marshmallow schema."""
+
+
+class DMSEndpointS3SettingsConfiguration(BaseSchema):
+    """DDK DMS Endpoint S3 Settings Marshmallow schema."""
+
+
+class DMSFactory:
+    """
+    Class factory create and configure Kinesis DDK resources, including Delivery Streams.
+    """
+
+    @staticmethod
+    def endpoint_settings_s3(
+        scope: Construct,
+        id: str,
+        environment_id: str,
+        bucket_name: str,
```

Comment:
Can we add prefix here as well? In case ingesting into the shared raw bucket

Reply:
added `bucket_folder`